logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/MessageSendFailureHandler.kt

```diff
@@ -34,11 +39,27 @@ interface MessageSendFailureHandler {
      * @return Either.Right if the error was properly handled and a new attempt at sending message can be made
      */
     suspend fun handleClientsHaveChangedFailure(sendFailure: ProteusSendMessageFailure): Either<CoreFailure, Unit>
+
+    /**
+     * Handle a failure when attempting to send a message
+     * update the message status to FAILED or FAILED_REMOTELY depending on the resulted failure.
+     * @param failure the failure that occured
+     * @param conversationId id of the conversation of the message that failed
+     * @param messageId id of the message that failed
+     * @param messageType type of the message that failed (for logging purposes)
+     */
+    suspend fun handleFailureUpdateMessageStatus(
```

Comment:
at first look, I thought this method was handling an update event that had failed 🙈  
What about `handleFailedMessageDelivery` or `handleMessageFailureAndUpdateStatus` if we want to specify more concretely what action we are taking?

Naming is hard 😅

Reply:
You're right, I didn't know which name to choose, I changed it to `handleFailureAndUpdateMessageStatus`, thanks!